src/store/models/network.ts

```diff
@@ -728,6 +757,60 @@ const networkModel: NetworkModel = {
       return network;
     },
   ),
+  setAutoMineMode: action((state, { id, mode }) => {
+    const network = state.networks.find(n => n.id === id);
+    if (!network) throw new Error(l('networkByIdErr', { id }));
+
+    network.autoMineMode = mode;
+  }),
+  setMiningState: action((state, { id, mining }) => {
+    state.autoMiners[id].mining = mining;
+  }),
+  mineBlock: thunk(async (actions, { id }, { getStoreState, getStoreActions }) => {
+    const { networks } = getStoreState().network;
+    const network = networks.find(n => n.id === id);
+    if (!network) throw new Error(l('networkByIdErr', { id }));
+
+    const { bitcoind } = getStoreActions();
+    const node = network.nodes.bitcoin[0];
+
+    actions.setMiningState({ id, mining: true });
+
+    bitcoind.mine({ node, blocks: 1 }).finally(() => {
+      bitcoind.getInfo(node);
+      actions.setMiningState({ id, mining: false });
+    });
+  }),
+  autoMine: thunk(async (actions, { id, mode, networkLoading }, { getState }) => {
+    const { networks, autoMiners } = getState();
+    const network = networks.find(n => n.id === id);
+    if (!network) throw new Error(l('networkByIdErr', { id }));
+
+    if (!autoMiners[id]) {
+      autoMiners[id] = {
+        mining: false,
+        startTime: 0,
+        timer: 0,
+      };
+    }
+
+    if (networkLoading || network.autoMineMode !== mode) {
+      await actions.save();
+
+      if (mode !== AutoMineMode.AutoOff) {
+        autoMiners[id].startTime = Date.now();
+        autoMiners[id].timer = +setInterval(
+          () => actions.mineBlock({ id }),
+          1000 * +mode,
+        );
```

Comment:
The thing with AutoMineMode is, that although the enum values might very well be a 'number' that would force us to coerce it elsewhere to a string in ant's selectedKeys property (and corresponding menu keys mapping) since it required an array of string values

![image](https://user-images.githubusercontent.com/15138241/202145951-2628e6a4-b648-40ff-9c35-9186af0b63ea.png)

Reply:
I see. This is a common issue with UI component libraries. From my experience, I find it best to use the optimal type in the application logic, to make this layer as clean as possible. Then perform the conversion only when setting the props on the UI component, since that is where the incompatibility lies.